urlpattern/resources/urlpatterntestdata.json

```diff
@@ -2367,15 +2375,24 @@
   },
   {
     "pattern": [{ "hostname": "bad#hostname" }],
-    "expected_obj": "error"
+    "exactly_empty_components": ["port"],
+    "expected_match": {
+      "hostname": { "input": "bad", "groups": {} }
+    }
   },
   {
     "pattern": [{ "hostname": "bad%hostname" }],
-    "expected_obj": "error"
+    "exactly_empty_components": ["port"],
```

Comment:
It seems I forgot to add inputs key to all of these new tests.

Reply:
Also see https://github.com/whatwg/urlpattern/issues/260.